skimage/measure/_regionprops.py

```diff
@@ -358,22 +358,73 @@ def __eq__(self, other):
         return True
 
 
+def to_dict(regions, want=['label', 'bbox'], separator='-', cancel_auto=False):
+    """Convert image regions properties into a dictionary
+
+    Parameters
+    ----------
+    regions : (N,) list
+        _RegionProperties
+    want : list, optional
+        str of a property the user wants to include
+    separator : str, optional
+        used in joining coordinates
+    cancel_auto : bool, optional
+        prevents autmatic adding of label and bbox
+
+
+    Output
+    ------
+    A dictionary with properties in accordance to the wanted properties
+
+    Examples
+    --------
+    >>> from skimage import data, util
+    >>> from skimage.measure import label
+    >>> from pandas.DataFrame import from_dict
+    >>> img = util.img_as_ubyte(data.coins()) > 110
+    >>> label_img = label(img, connectivity=img.ndim)
+    >>> props = regionprops(label_img)
+    >>> out = from_dict(to_dict(props))
+
+
+"""
+
+    objects = {'image', 'coords', 'convex_image', 'filled_image',
+               'intensity_image'}
+
+    if not cancel_auto and 'label' not in want:
+        want.insert(0, 'label')
+    if not cancel_auto and 'bbox' not in want:
+        want.insert(0, 'bbox')
+
+    out = {}
+    arr = len(regions) * [0]
+    for prop in want:
+        r = regions[0][prop]
+        if np.isscalar(r) or prop in objects:
+            out[prop] = [regions[i][prop] for i in range(len(arr))]
+        else:
+            if isinstance(r, np.ndarray):
+                shape = r.shape
+            else:
+                shape = (len(r),)
+
+            for ind in np.ndindex(shape):
+                for k in range(len(arr)):
+                    arr[k] = regions[k][prop][ind if len(ind) > 1 else ind[0]]
+                out[separator.join(map(str, (prop,) + ind))] = arr[:]
```

Comment:
I think it's better to use `np.copy(arr)`, because it is more explicit.

I might also suggest using the name `column_buffer` rather than `column` or `arr`, because it makes it clear that we know it's temporary.